src/main/java/decodes/cwms/CwmsTimeSeriesDAO.java

```diff
@@ -1407,9 +1414,16 @@ public DataCollection getNewData(DbKey applicationId)
 			ex.printStackTrace();
 			throw new DbIoException("Error reading new data: " + ex);
 		}
-		finally
+	}
+
+	final boolean exceedsMaxTimeGap(Date lastTimestamp, Date currentTimestamp)
+	{
+		if(lastTimestamp == null || currentTimestamp == null)
 		{
+			return false;
 		}
+		long daysBetween = TimeUnit.MILLISECONDS.toDays(currentTimestamp.getTime() - lastTimestamp.getTime());
+		return daysBetween > (DecodesSettings.instance().cp_cwmstsdb_getNewData_max_timegap_days - 1);
```

Comment:
wouldn't this just be `>=` or am I mixing up direction?

Reply:
Yup, you're right. Much easier to read.